charts/datadog/templates/confd-configmap.yaml

```diff
@@ -20,4 +21,10 @@ override duplicates
 {{- if .Values.datadog.confd }}
 {{ tpl (toYaml .Values.datadog.confd) . | indent 2 }}
 {{- end }}
+{{- range $integration, $configs := $.Values.datadog.advancedConfd }}
+{{- range $name, $config := $configs }}
+  {{ printf "%s--%s: |" $integration $name }}
```

Comment:
I just tested this out and instead of having this nested structure: `cilium.d/cilium-agent.yaml`, it outputs a file with this filename `cilium.d/cilium-agent.yaml`.

This should do the former instead.